Clarify code around 32 bits OMF compression of symbols

Geod24 opened the pull request:
~Thanks to the new mangling scheme, it shouldn't be needed anymore.~

```
While looking at the stack-trace code, I came accross 'decodeDmdString',
which seemed like it was no longer needed. After a comment from Rainers,
it turned out to still be required for OMF builds, so this adds comments
in all places where this appears for the next reader, as well as make
the decodeDmdString function package so that it doesn't show up in documentation.
```

rainers commented:
As long as there is OMF support, I would recommend to keep it. Symbols of length longer than 128 are still compressed, see https://github.com/dlang/dmd/blob/master/compiler/src/dmd/backend/cgobj.d#L2442.

Geod24 commented:
Thanks, I'll change the PR to document this then.

thewilsonator commented:
>As long as there is OMF support 

Speaking of which, when is that getting the axe?

Geod24 committed:
Clarify code around 32 bits OMF compression of symbols

While looking at the stack-trace code, I came accross 'decodeDmdString',
which seemed like it was no longer needed. After a comment from Rainers,
it turned out to still be required for OMF builds, so this adds comments
in all places where this appears for the next reader.

Geod24 commented:
Reworked the PR to document things.

rainers reviewed (changes requested):
Sorry, just remembered that I've been using `decodeDmdString` in a couple of tools, including Visual D. Making it `package` will break that, so it is likely others might be affected, too.

Can it still be public without being added to the documentation?

Geod24 commented:
Sure thing, fixed
